webapp/controller/ListReport.controller.js

```diff
@@ -295,23 +235,39 @@ sap.ui.define([
 			const oComponent = this.getOwnerComponent();
 			
 			oComponent.getRouter().navTo("ProductDetails", {
-				productId: oCtx.getObject("Id")
+				productId: oCtx.getObject("ID"),
+				mode: "view"
 			})
 		},
 
 		onCreateButtonPress: function() {
 			const sProductId = "newProduct";
 			const oComponent = this.getOwnerComponent();
 			oComponent.getRouter().navTo("ProductDetails", {
-				productId: sProductId
+				productId: sProductId,
+				mode: "create"
 			})	
 		},
-		
-		rewriteProductsIds: function() {
-			const aProducts = this.getModel().getProperty("/Products");
-			aProducts.forEach((el, id) => {
-				el.Id = String(id + 1);
-				return el;
+
+		onTableSelectionChange: function(oEvent) {
+			const oViewModel = this.getModel("view");
+			const aSelectedItems = oEvent.getSource().getSelectedItems();
+
+			oViewModel.setProperty("/isButtonEnable", !!aSelectedItems.length)
+		},
+
+		deleteProducts: function() {		
+			const oTable = this.byId("idProductsTable");
+			const oODataModel = this.getModel();
+			const aSelectedItemsIds = oTable.getSelectedItems().map(el => el.getBindingContext().getObject("ID"));
+			aSelectedItemsIds.forEach(el => {
+				const sKey = oODataModel.createKey("/Products", {ID: el});
+				oODataModel.remove(sKey, {
+					success: () => {
```

Comment:
You need to handle error case. 
Always do that when you invoke CRUD operation or something might fail.